src/audio/module_adapter/module/volume/volume_hifi3_with_peakvol.c

```diff
@@ -440,7 +441,7 @@ static void vol_s24_to_s24_s32(struct processing_module *mod, struct input_strea
 				/* Store the output sample */
 				AE_S32_L_XP(out_sample, out, inc);
 			}
-			peak_vol = AE_SLAA32S(peak_vol, attenuation);
+			peak_vol = AE_SLAA32S(peak_vol, attenuation + PEAK_24S_32C_ADJUST);
```

Comment:
attenuation normally means >>, not <<,  used to avoid saturation
     attenuation + PEAK_24S_32C_ADJUST? or PEAK_24S_32C_ADJUST - attenuation?

its value normally 0, 1, 2

Reply:
@btian1 volume module use attenuation to compensate signal and report correct volume in Peak Meter regs which is previously attenuated by host copier. Thus we need to add attenuation to shift left Peak Meter value accordingly.